src/Docker/Compose/DeveloperCompose.php

```diff
@@ -30,7 +30,8 @@ public function build(Repository $config): array
     {
         $compose = parent::build($config);
         $compose['volumes'] = [
-            'magento-sync' => self::SYNC_ENGINE_DOCKER_SYNC === $config[self::SYNC_ENGINE] ? ['external' => true] : []
+            'magento-sync' => self::SYNC_ENGINE_DOCKER_SYNC === $config->get(self::SYNC_ENGINE) ?
```

Comment:
I personally think this format's a lot easier to read across three lines:
```php
'magento-sync' => self::SYNC_ENGINE_DOCKER_SYNC === $config[self::SYNC_ENGINE]
    ? ['external' => true]
    : []
```

but that's a matter of preference